internal/service/news/news.go

```diff
@@ -0,0 +1,144 @@
+	n.appCtx.WgAdd(1)
+	defer n.appCtx.WgDone()
+
+	refreshTime := cfg.Core.RefreshInterval
+	timer := time.NewTimer(refreshTime)
+
+	if err := n.getArticels(); err != nil {
+		err = fmt.Errorf("news-srv: failed to get articels %s", err)
+		log.Println(err)
+	}
+
+	for {
+		timer.Reset(refreshTime)
+		select {
+		case <-n.appCtx.GetGoContext().Done():
+			if !timer.Stop() {
+				<-timer.C
+			}
+			return
+		case <-timer.C:
+		}
+
+		if err := n.getArticels(); err != nil {
+			err = fmt.Errorf("news-srv: failed to get articels %s", err)
+			log.Println(err)
+		}
+	}
+}
+
+func (n *NewsServer) getArticels() error {
+	articlesExt := new(entity.ArticleListExternal)
+	if err := fetchData(articlesListEndpoint, articlesExt); err != nil {
+		return err
+	}
+
+	articlesDb := make([]*entity.NewsletterNewsItemDB, 0, len(articlesExt.ArticleList.Articles))
+	for _, article := range articlesExt.ArticleList.Articles {
+		articleDb := &entity.NewsletterNewsItemDB{
+			ArticleURL:        article.ArticleURL,
+			NewsArticleID:     article.NewsArticleID,
+			PublishDate:       article.PublishDate,
+			Taxonomies:        article.Taxonomies,
+			TeaserText:        article.TeaserText,
+			ThumbnailImageURL: article.ThumbnailImageURL,
+			Title:             article.Title,
+			OptaMatchId:       article.OptaMatchId,
+			LastUpdateDate:    article.LastUpdateDate,
+			IsPublished:       article.IsPublished,
+		}
+
+		articlesDb = append(articlesDb, articleDb)
+	}
+	insertedArticlesId, err := n.article.SaveArticleList(n.appCtx.GetGoContext(), articlesDb)
+	if err != nil {
+		return err
+	}
+
+	for _, id := range insertedArticlesId {
+		articleExt := new(entity.ArticleExternal)
+		if err := fetchData(fmt.Sprintf(articleEndpoint, id), articleExt); err != nil {
+			return err
+		}
+
+		articleDb := &entity.NewsArticleDB{
+			ArticleURL:        articleExt.NewsArticle.ArticleURL,
+			NewsArticleID:     articleExt.NewsArticle.NewsArticleID,
+			PublishDate:       articleExt.NewsArticle.PublishDate,
+			Taxonomies:        articleExt.NewsArticle.Taxonomies,
+			TeaserText:        articleExt.NewsArticle.TeaserText,
+			Subtitle:          articleExt.NewsArticle.Subtitle,
+			ThumbnailImageURL: articleExt.NewsArticle.ThumbnailImageURL,
+			Title:             articleExt.NewsArticle.Title,
+			BodyText:          articleExt.NewsArticle.BodyText,
+			GalleryImageURLs:  articleExt.NewsArticle.GalleryImageURLs,
+			VideoURL:          articleExt.NewsArticle.VideoURL,
+			OptaMatchId:       articleExt.NewsArticle.OptaMatchId,
+			LastUpdateDate:    articleExt.NewsArticle.LastUpdateDate,
+			IsPublished:       articleExt.NewsArticle.IsPublished,
+		}
+
+		if _, err := n.article.SaveArticle(n.appCtx.GetGoContext(), articleDb); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func fetchData(url string, data any) error {
+	resp, err := http.Get(url)
+	if err != nil {
+		return fmt.Errorf("news-service: failed to get response: %s", err)
+	}
+
```

Comment:
`defer resp.Body.Close()` missing - https://youtu.be/a6MARo70YBg

Reply:
fixed

Reply:
fixed